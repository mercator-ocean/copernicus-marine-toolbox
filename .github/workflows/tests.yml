name: Run tests

on:
  pull_request:
    branches: [ "main", "copernicusmarine-toolbox-v2" ]

jobs:
  tests:
    runs-on: self-hosted
    timeout-minutes: 30

    steps:
      - name: Check out code
        uses: actions/checkout@v4

      - uses: mamba-org/setup-micromamba@v1
        with:
          micromamba-version: '1.5.6-0'
          micromamba-binary-path: ${{ runner.temp }}/bin/micromamba
          environment-file: conda_environment_test.yaml
          environment-name: copernicusmarine_test
          condarc-file: .condarc
          cache-environment: true
          post-cleanup: 'all'

      - name: Run tests
        env:
          COPERNICUSMARINE_SERVICE_USERNAME: ${{ secrets.COPERNICUSMARINE_SERVICE_USERNAME }}
          COPERNICUSMARINE_SERVICE_PASSWORD: ${{ secrets.COPERNICUSMARINE_SERVICE_PASSWORD }}
        run: make run-tests
        shell: micromamba-shell {0}
