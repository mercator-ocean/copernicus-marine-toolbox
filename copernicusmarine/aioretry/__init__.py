from .retry import BeforeRetry  # noqa: F401
from .retry import RetryInfo  # noqa: F401
from .retry import RetryPolicy  # noqa: F401
from .retry import RetryPolicyStrategy  # noqa: F401
from .retry import retry  # noqa: F401
