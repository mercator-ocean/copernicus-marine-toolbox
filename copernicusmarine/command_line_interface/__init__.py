# from copernicusmarine.command_line_interface.group_describe import describe

# group_describe = describe
