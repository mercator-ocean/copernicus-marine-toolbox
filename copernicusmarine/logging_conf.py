logging_configuration_dict = {
    "disable_existing_loggers": False,
    "formatters": {
        "blank": {"format": "%(message)s"},
        "simple": {
            "datefmt": "%Y-%m-%dT%H:%M:%SZ",
            "format": "%(levelname)s - %(asctime)s - %(message)s",
        },
    },
    "handlers": {
        "console": {
            "class": "logging.StreamHandler",
            "formatter": "simple",
            "level": "DEBUG",
            "stream": "ext://sys.stderr",
        },
        "console_blank": {
            "class": "logging.StreamHandler",
            "formatter": "blank",
            "level": "DEBUG",
            "stream": "ext://sys.stdout",
        },
    },
    "loggers": {
        "copernicus_marine_blank_logger": {
            "handlers": ["console_blank"],
            "level": "INFO",
            "propagate": False,
        },
        "copernicus_marine_root_logger": {
            "handlers": ["console"],
            "level": "INFO",
        },
    },
    "version": 1,
}
