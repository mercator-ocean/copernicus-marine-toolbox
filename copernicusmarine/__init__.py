"""
.
"""

import logging.config
import time
from importlib.metadata import version

from copernicusmarine.logging_conf import logging_configuration_dict

__version__ = version("copernicusmarine")

logging.config.dictConfig(logging_configuration_dict)
logging.Formatter.converter = time.gmtime

from copernicusmarine.python_interface.describe import describe
from copernicusmarine.python_interface.get import get
from copernicusmarine.python_interface.login import login
from copernicusmarine.python_interface.open_dataset import (
    load_xarray_dataset,  # depracated
    open_dataset,
)
from copernicusmarine.python_interface.read_dataframe import (
    load_pandas_dataframe,  # depracated
    read_dataframe,
)
from copernicusmarine.python_interface.subset import subset
