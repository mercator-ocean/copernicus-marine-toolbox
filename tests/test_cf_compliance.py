import json

import xarray

from copernicusmarine import subset
from tests.test_utils import execute_in_terminal


class TestCFCompliance:
    def test_subset_open(self, tmp_path, snapshot):
        dataset_id = "cmems_mod_nws_bgc-pft_my_7km-3D-pico_P1M-m"
        self.if_I_subset_a_dataset(dataset_id, tmp_path, "output_1.nc", "pico")
        self.then_it_is_cf_compliant(
            dataset_id, tmp_path, snapshot, "output_1"
        )

    def test_subset_with_warns(self, tmp_path, snapshot):
        dataset_id = (
            "cmems_obs-sst_med_phy-sst_nrt_diurnal-oi-0.0625deg_PT1H-m"
        )
        self.if_I_subset_a_dataset(
            dataset_id,
            tmp_path,
            "output_2.nc",
            "analysed_sst",
        )
        self.then_it_is_cf_compliant(
            dataset_id, tmp_path, snapshot, "output_2"
        )

    def if_I_subset_a_dataset(
        self, dataset_id, tmp_path, output_filename, variable
    ):
        subset(
            dataset_id=dataset_id,
            variables=[variable],
            output_directory=tmp_path,
            output_filename=output_filename,
            start_datetime="2022-01-01T00:00:00",
            end_datetime="2022-01-05T00:00:00",
        )
        assert (tmp_path / output_filename).exists()

    def then_it_is_cf_compliant(
        self, dataset_id, tmp_path, snapshot, output_filename
    ):
        dataset_id = dataset_id
        dataset = xarray.open_dataset(f"{tmp_path}/{output_filename}.nc")
        CF_convention = dataset.attrs["Conventions"][-3:]
        if CF_convention < "1.6":
            CF_convention = "1.6"
        command = [
            "compliance-checker",
            f"--test=cf:{CF_convention}",
            f"{tmp_path}/{output_filename}.nc",
            "-f",
            "json",
            "-o",
            f"{tmp_path}/{output_filename}_checked.json",
        ]
        self.output = execute_in_terminal(command)
        assert self.output.returncode == 0

        f = open(f"{tmp_path}/{output_filename}_checked.json")
        data = json.load(f)

        list_msgs = []
        for diccionari in data[f"cf:{CF_convention}"]["all_priorities"]:
            if len(diccionari["msgs"]) > 0:
                list_msgs.append(diccionari["name"])
                list_msgs.append(diccionari["msgs"])

        assert dataset_id == snapshot
        assert data[f"cf:{CF_convention}"]["scored_points"] == snapshot
        assert data[f"cf:{CF_convention}"]["possible_points"] == snapshot
        assert list_msgs == snapshot
