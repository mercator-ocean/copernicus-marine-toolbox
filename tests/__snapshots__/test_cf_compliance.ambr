# serializer version: 1
# name: TestCFCompliance.test_subset_open
  'cmems_mod_nws_bgc-pft_my_7km-3D-pico_P1M-m'
# ---
# name: TestCFCompliance.test_subset_open.1
  160
# ---
# name: TestCFCompliance.test_subset_open.2
  160
# ---
# name: TestCFCompliance.test_subset_open.3
  list([
  ])
# ---
# name: TestCFCompliance.test_subset_with_warns
  'cmems_obs-sst_med_phy-sst_nrt_diurnal-oi-0.0625deg_PT1H-m'
# ---
# name: TestCFCompliance.test_subset_with_warns.1
  135
# ---
# name: TestCFCompliance.test_subset_with_warns.2
  136
# ---
# name: TestCFCompliance.test_subset_with_warns.3
  list([
    'ยง2.6 Attributes',
    list([
      'ยง2.6.1 Conventions global attribute does not contain "CF-1.6"',
    ]),
  ])
# ---
